docs/indepth/installation.rst

```diff
@@ -324,6 +328,72 @@ Setup the pre-requisite repositories and install the ``clusterhq-flocker-node``
 .. task:: install_flocker ubuntu-14.04
    :prompt: [root@ubuntu]#
 
+.. _authentication:
+
+Cluster Authentication Layer Configuration
+------------------------------------------
+
+Communication between the different parts of your cluster is secured and authenticated via TLS.
+The Flocker CLI package includes the ``flocker-ca`` tool that is used to generate TLS certificate and key files that you will need to copy over to your nodes.
+
+Once you have installed the ``flocker-node`` package, you will need to generate:
+
+- A control service certificate and key file, to be copied over to the node running your :doc:`control service <../advanced/architecture>`.
+- A certificate and key file for each of your nodes, which you will also need to copy over to the nodes.
+
+Both types of certificate will be signed by a certificate authority identifying your cluster, which is also generated using the ``flocker-ca`` tool.
+
+Using the machine on which you installed the ``flocker-cli`` package, run the following command to generate your cluster's root certificate authority, replacing ``mycluster`` with any name you like to uniquely identify this cluster.
+
+.. code-block:: console
+
+    $ flocker-ca initialize mycluster
+    Created cluster.key and cluster.crt. Please keep cluster.key secret, as anyone who can access it will be able to control your cluster.
+
+You will find the files ``cluster.key`` and ``cluster.crt`` have been created in your working directory.
+The file ``cluster.key`` should be kept only by the cluster administrator; it does not need to be copied anywhere.
+
+You are now able to generate authentication certificates for the control service and each of your nodes.
+To generate the control service certificate, run the following command from the same directory containing your authority certificate generated in the previous step.
+
+.. code-block:: console
+
+   $ flocker-ca create-control-certificate
+   Created control-service.crt. Copy it over to /etc/flocker/control-service.crt on your control service machine and make sure to chmod 0600 it.
+   
+You will need to copy both ``control-service.crt`` and ``control-service.key`` over to the node that is running your control service, to the directory ``/etc/flocker/``.
+You should also copy the cluster's public certificate, the `cluster.crt` file.
+On the server, the `/etc/flocker` directory and private key file should be set to secure permissions via `chmod`:
+
+.. code-block:: console
+
+   root@mercury:~/$ chmod 600 /etc/flocker
+   root@mercury:~/$ chmod 600 /etc/flocker/control-service.key
+
+You should copy these files via a secure communication medium such as SSH, SCP or SFTP.
+
+.. warning::
+
+   For the cluster's certificate, copy the file ``cluster.crt`` **only** - do **not** copy the ``cluster.key`` file; this must kept only by the cluster administrator.
```

Comment:
Bit confusing, maybe "Only copy ``cluster.crt`` to the control service and node machines, not ``cluster.key``" or something.